doc/design/staging-rename.md

```diff
@@ -0,0 +1,24 @@
+# Rename "staging" to "sandbox"
+
+We have long referred to our test deployment of the DANDI Archive as “staging”. This is due to our deployment workflow, which auto-deploys whatever is on `master` to https://gui-staging.dandiarchive.org, allowing us to test features in a production-like environment before they are deployed to actual production. We have additionally advertised this deployment to archive users as a way to try things out in a safe area before doing real work on the production instance.
+
+The latter usage has led to user confusion about the role of the staging deployment. Specifically, the name “staging” makes people think that they are supposed to put data here for review and cleaning, etc., before it makes its way to a more permanent home somewhere.
+
+To avoid that confusion, we are changing the name of the this deployment from “staging” to “sandbox”, better reflecting the end user experience of this deployment. Despite the name change, we will continue to use our deployment strategy without any changes.
+
+## Requirements
+
+The following changes will be required to make this name change happen.
+
+1. **New URL for sandbox deployment.** The deployment formerly known as “staging” will be accessible at `https://sandbox.dandiarchive.org`.
+2. **New URL for sandbox API.** The corresponding Django API app will be accessible at `https://api.sandbox.dandiarchive.org`.
+3. **Redirections for old URLs.** To prevent user pain, we will redirect requests to the old application and API URLs to the new ones:
+    1. [`https://gui-staging.dandiarchive.org`](https://gui-staging.dandiarchive.org) → `https://sandbox.dandiarchive.org`
+    2. [`https://api-staging.dandiarchive.org`](https://api-staging.dandiarchive.org) → [`https://api.sandbox.dandiarchive.org`](https://api.sandbox.dandiarchive.org) (using HTTP 308 to preserve the request method)
+
+    The second redirection is probably less important for end users (excluding those that have written, e.g., Python scripts using the DANDI API against staging/sandbox for some reason), but it is necessary to avoid deprecating all previous versions of the DANDI CLI tool which still presumably references the old URLs.
```

Comment:
@yarikoptic, @jwodder, could you confirm the claim about deprecating earlier versions of dandi-cli?

Reply:
@waxlamp I'm unclear what you're asking.  Of course old versions of `dandi-cli` point to api-staging.  What else would they do?

Reply:
I should have been clearer. I was assuming that you would *not* want to deprecate all previous versions of dandi-cli just to accommodate a change in the URLs, and that's what I was asking for confirmation on.

Reply:
i would be ok with deprecating prior cli versions when this change happens.

Reply:
My 1c would be -- to not bother. If someone runs into an issue, an obvious solution would be to upgrade. But as not everyone is using staging, and client would work perfectly fine with the main and other instances (and actually for downloads should work also with renamed one) -- we should be all good.

For the longer run eventually we should likely address
- https://github.com/dandi/dandi-cli/issues/1522

Reply:
> (and actually for downloads should work also with renamed one) 

It will _not_ work for sandbox unless we redirect api-staging to api.sandbox, right? (That's what's at issue here, whether we need to put that redirection in place. But I think we should just do so.)